src/parser/transform/statement/transform_update.cpp

```diff
@@ -19,8 +19,13 @@ unique_ptr<UpdateSetInfo> Transformer::TransformUpdateSetInfo(duckdb_libpgquery:
 
 unique_ptr<UpdateStatement> Transformer::TransformUpdate(duckdb_libpgquery::PGUpdateStmt &stmt) {
 	auto result = make_uniq<UpdateStatement>();
+	vector<unique_ptr<CTENode>> materialized_ctes;
 	if (stmt.withClause) {
-		TransformCTE(*PGPointerCast<duckdb_libpgquery::PGWithClause>(stmt.withClause), result->cte_map);
+		TransformCTE(*PGPointerCast<duckdb_libpgquery::PGWithClause>(stmt.withClause), result->cte_map,
+		             materialized_ctes);
+		if (!materialized_ctes.empty()) {
+			throw NotImplementedException("Materialized CTEs are not implemented for delete.");
```

Comment:
Should this be `update` - also could you perhaps expand on the reason for this?

Reply:
You are right, the message is wrong. I will fix that.

---

There is an AST 'mismatch' currently. Materialized CTEs are represented as a `QueryNode`, whereas`INSERT/UPDATE/DELETE` are `SQLStatements`. This means, that it is easy to represent the materialized CTEs as a bunch of stacked `QueryNodes` (this happens in `Transformer::TransformMaterializedCTE`). It is not that easy for `INSERT/UPDATE/DELETE`, however. While there should not be a principle problem implementing that, my current plan is to tackle that in a separate pr.